public/app/core/components/AppChrome/History/HistoryWrapper.tsx

```diff
@@ -0,0 +1,204 @@
+import { css } from '@emotion/css';
+import moment from 'moment';
+import { useState } from 'react';
+
+import { FieldType, GrafanaTheme2, store } from '@grafana/data';
+import { Button, Card, IconButton, Space, Stack, Text, useStyles2, Box, Sparkline, useTheme2 } from '@grafana/ui';
+import { t } from 'app/core/internationalization';
+
+import { HISTORY_LOCAL_STORAGE_KEY } from '../AppChromeService';
+import { HistoryEntry } from '../types';
+
+export function HistoryWrapper({ onClose }: { onClose: () => void }) {
+  const history = store.getObject<HistoryEntry[]>(HISTORY_LOCAL_STORAGE_KEY, []).filter((entry) => {
+    return moment(entry.time).isAfter(moment().subtract(2, 'day').startOf('day'));
+  });
+  const [numItemsToShow, setNumItemsToShow] = useState(5);
+
+  const selectedTime = history.find((entry) => {
+    return entry.url === window.location.href || entry.views.some((view) => view.url === window.location.href);
+  })?.time;
+
+  const hist = history.slice(0, numItemsToShow).reduce((acc: { [key: string]: HistoryEntry[] }, entry) => {
+    const date = moment(entry.time);
+    let key = '';
+    if (date.isSame(moment(), 'day')) {
+      key = t('nav.history-wrapper.today', 'Today');
+    } else if (date.isSame(moment().subtract(1, 'day'), 'day')) {
+      key = t('nav.history-wrapper.yesterday', 'Yesterday');
+    } else {
+      key = date.format('YYYY-MM-DD');
+    }
+    acc[key] = [...(acc[key] || []), entry];
+    return acc;
+  }, {});
```

Comment:
@JoaoSilvaGrafana this is pretty messy / complex code, I would really prefer some state / store abstractions on top of the local storage history. reading and updating locale storage object for history in multiple places and dealing with the state logic of that store state directly in the render code or in AppChromeService seems not ideal